src/run_solver.jl

```diff
@@ -19,6 +19,10 @@ Apply a solver to a set of problems.
   (default: `x->false`);
 * `colstats::Vector{Symbol}`: summary statistics for the logger to output during the
 benchmark (default: `[:name, :nvar, :ncon, :status, :elapsed_time, :objective, :dual_feas, :primal_feas]`);
+* `solver_specific`: scalar entries in a solver's `s.solver_specific` are
+  appended as extra columns to the returned `DataFrame` (vector entries are
+  ignored); columns are created when first encountered and earlier skipped/exception
+  rows contain `missing` for these columns.
```

Comment:
```suggestion
* `colstats::Vector{Symbol}`: summary statistics for the logger to output during the
benchmark (default: `[:name, :nvar, :ncon, :status, :elapsed_time, :objective, :dual_feas, :primal_feas]`). Solver's `solver_specific` scalar entries are appended as extra columns.
```

Have you tested doing ? solve_problems in the REPL to see how it looks?

Reply:
I ran solve_problems with the CallableSolver and got a non-exception result.

What I ran (repro)
cd /home/arnav/Desktop/SolverBenchmark.jljulia --project=@. e 'import Pkg; Pkg.instantiate(); using SolverBenchmark, SolverCore, NLPModels, ADNLPModels; struct CallableSolver end; function (solver::CallableSolver)(nlp::AbstractNLPModel; kwargs...) return GenericExecutionStats(nlp) end; problems = [ADNLPModel(x -> sum(x .^ 2), ones(2), name = "Quadratic")]; stats = solve_problems(CallableSolver(), "callable", problems); println(stats); println("--- columns: ", names(stats))'

Key results
- The returned DataFrame is 1×40 and does not have status = :exception.
- The row shows status == :unknown (from the GenericExecutionStats returned by the callable) and the usual columns are present.
- Columns printed:
["solver_name", "id", "name", "nvar", "ncon", "nequ", "status", "objective", "elapsed_time", "iter", "dual_feas", "primal_feas", "neval_obj", "neval_grad", "neval_cons", "neval_cons_lin", "neval_cons_nln", "neval_jcon", "neval_jgrad", "neval_jac", "neval_jac_lin", "neval_jac_nln", "neval_jprod", "neval_jprod_lin", "neval_jprod_nln", "neval_jtprod", "neval_jtprod_lin", "neval_jtprod_nln", "neval_hess", "neval_hprod", "neval_jhess", "neval_jhprod", "neval_residual", "neval_jac_residual", "neval_jprod_residual", "neval_jtprod_residual", "neval_hess_residual", "neval_jhess_residual", "neval_hprod_residual", "extrainfo"]